src/lib/component/view/ChecksView.tsx

```diff
@@ -0,0 +1,189 @@
+import * as React from "/vendor/react";
+import gql from "/vendor/graphql-tag";
+import { ApolloError } from "/vendor/apollo-client";
+
+import {
+  parseFilterParams,
+  buildFilterParams,
+  FilterMap,
+} from "/lib/util/filterParams";
+import { pollingDuration } from "/lib/constant/polling";
+import { FailedError } from "/lib/error/FetchError";
+
+import {
+  parseIntParam,
+  parseStringParam,
+  parseArrayParam,
+} from "/lib/util/params";
+
+import {
+  useSearchParams,
+  useQuery,
+  UseQueryResult,
+  WithWidth,
+  useRouter,
+} from "/lib/component/util";
+import {
+  MobileFullWidthContent,
+  Content,
+  FilterList,
+} from "/lib/component/base";
+import {
+  AppLayout,
+  ChecksList,
+  ChecksListToolbar,
+  NotFound,
+} from "/lib/component/partial";
+
+const Query = gql`
+  query ChecksViewQuery(
+    $namespace: String!
+    $limit: Int
+    $offset: Int
+    $order: CheckListOrder
+    $filters: [String!]
+  ) {
+    namespace(name: $namespace) {
+      ...ChecksList_namespace
+    }
+  }
+
+  ${ChecksList.fragments.namespace}
+`;
+
+interface Variables {
+  namespace: string;
+  limit: number;
+  offset: number;
+  order: string;
+  filters: string[];
+}
+
+export function useChecksViewQuery(variables: Variables) {
```

Comment:
I realize this reduces a bit of duplicated code, but I was imagining that it would be preferable to have the upstream Component compose it's own query. I guess passing down the variables would be a pain though. 🤔

Reply:
I agree with you on that. Stay posted.